ui/common/src/socket.ts

```diff
@@ -289,57 +283,65 @@ export default class StrongSocket {
     const ws = this.ws;
     if (ws) {
       this.debug('Disconnect');
-      this.autoReconnect = false;
       ws.onerror = ws.onclose = ws.onopen = ws.onmessage = () => {};
       ws.close();
     }
   };
 
-  onError = (e: unknown): void => {
+  private onError = (e: unknown): void => {
+    if (this.heartbeat.wasSuspended) return;
     this.options.debug = true;
     this.debug(`error: ${e} ${JSON.stringify(e)}`); // e not always from lila
   };
 
-  onClose = (e: CloseEvent, url: string): void => {
+  private onClose = (e: CloseEvent): void => {
     site.pubsub.emit('socket.close');
-    if (this.autoReconnect) {
+
+    if (this.heartbeat.wasSuspended) return this.onSuspended();
+    this.storeStats({ event: 'close', code: e.code });
+
+    if (this.ws) {
       this.debug('Will autoreconnect in ' + this.options.autoReconnectDelay);
       this.scheduleConnect(this.options.autoReconnectDelay);
     }
     if (e.wasClean && e.code < 1002) return;
 
-    if (isOnline())
-      log(`${site?.sri ? 'sri ' + site.sri : ''} unclean close ${e.code} ${url} ${e.reason}`);
-    this.tryOtherUrl = true;
+    if (isOnline()) this.tryOtherUrl = true;
     clearTimeout(this.pingSchedule);
   };
 
-  onSuccess = (): void => {
-    this.nbConnects++;
-    if (this.nbConnects == 1) {
-      site.pubsub.complete('socket.connect');
-      let disconnectTimeout: Timeout | undefined;
-      idleTimer(
-        10 * 60 * 1000,
-        () => {
-          this.options.idle = true;
-          disconnectTimeout = setTimeout(this.destroy, 2 * 60 * 60 * 1000);
-        },
-        () => {
-          this.options.idle = false;
-          if (this.ws) clearTimeout(disconnectTimeout);
-          else location.reload();
-        },
-      );
-    }
+  private onSuccess = (): void => {
+    if (site.pubsub.past('socket.hasConnected')) return;
+
+    site.pubsub.complete('socket.hasConnected');
+    let disconnectTimeout: Timeout | undefined;
+    idleTimer(
+      10 * 60 * 1000,
+      () => {
+        this.options.idle = true;
+        disconnectTimeout = setTimeout(this.destroy, 2 * 60 * 60 * 1000);
+      },
+      () => {
+        this.options.idle = false;
+        if (this.ws) clearTimeout(disconnectTimeout);
+        else location.reload();
+      },
+    );
   };
 
-  baseUrl = (): string => {
+  private onSuspended() {
+    this.heartbeat.reset(); // not a networking error, just get our connection back
+    clearTimeout(this.pingSchedule);
+    clearTimeout(this.connectSchedule);
+    this.storeStats({ event: 'suspend' }).then(this.connect);
+  }
```

Comment:
this method is run when after a previously loaded page that was suspended becomes active again. we arrive here from `close` events that are generated for the websocket by the browser on activation.